src/masternodeman.cpp

```diff
@@ -1645,6 +1645,11 @@ void CMasternodeMan::SetMasternodeLastPing(const CTxIn& vin, const CMasternodePi
         return;
     }
     pMN->lastPing = mnp;
+    // if masternode uses sentinel ping instead of watchdog
+    // we shoud update nTimeLastWatchdogVote here if sentinel
+    // ping flag is actual
+    if(mnp.sentinelPing)
+        pMN->UpdateWatchdogVoteTime();
```

Comment:
Yes, this should be called from CMasternodeMan::ProcessMessage or in some function called from there.